### PR TITLE
Use default build script names

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

**Review by lptr — APPROVED** [review]

File: settings.gradle.kts [hunk 1]
```diff
@@ -168,7 +168,7 @@ val groovyBuildScriptProjects = hashSetOf(
 )
 
 fun buildFileNameFor(projectDirName: String) =
-    "$projectDirName${buildFileExtensionFor(projectDirName)}"
+    "build${buildFileExtensionFor(projectDirName)}"
```
Comment (jjohannes) [inline settings.gradle.kts:171]: :x: I think we should remove the `project.buildFileName` and all helper code for that completely, including the the _require_ stuff and the "groovyBuildScriptProjects" list (which only has one entry left anyway). And rely on the default behaviour of Gradle to discover the build scripts.

We should get rid of as much custom setting script build logic as possible.

**Review by jjohannes — CHANGES_REQUESTED** [review]
Thanks for following up so quickly.

I think we should remove all custom logic around the _build file names_ from settings. Only then will we experience what most users do.

File: settings.gradle.kts (hunk 1 above)
Comment (donat) [reply settings.gradle.kts]: Fixed.

**Comment by donat** [comment]
@bot-gradle test this

File: settings.gradle.kts [hunk 2]
```diff
@@ -161,27 +161,10 @@ val upperCaseLetters = "\\p{Upper}".toRegex()
...
     project.projectDir = file("subprojects/${project.name}")
-    project.buildFileName = buildFileNameFor(project.name)
-    require(project.projectDir.isDirectory) {
-        "Project directory ${project.projectDir} for project ${project.name} does not exist."
-    }
-    require(project.buildFile.isFile) {
-        "Build file ${project.buildFile} for project ${project.name} does not exist."
-    }
+    // TODO convert docs/build.gradle to Kotlin DSL https://github.com/gradle/gradle/issues/14393
+    project.buildFileName = "build.gradle${if (project.name == "docs") "" else ".kts"}"
```
Comment (jjohannes) [inline settings.gradle.kts:167]: We should not need this. Gradle will look for both "build.gradle" and "build.gradle.kts" on its own.

**Review by jjohannes — CHANGES_REQUESTED** [review]

File: settings.gradle.kts (hunk 2 above)
Comment (donat) [reply settings.gradle.kts]: Done

**Review by jjohannes — APPROVED** [review]
👍 Nice to see so much code from settings gone.

I noticed there is another line that can be removed, because the variable is not used:
`val upperCaseLetters = "\\p{Upper}".toRegex()`

Maybe you can remove that as well.

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
Closed via d2b854b.

**Pull request merged**